src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,92 @@
 package core.basesyntax;
 
-public class MyHashMap<K, V> implements MyMap<K, V> {
+import java.util.Objects;
+
+public class MyHashMap<K,V> implements MyMap<K,V> {
+    private static final int INITIAL_CAPACITY = 16;
+    private static final double LOAD_FACTOR = 0.75;
+    private Node<K,V>[] table;
+    private int size;
+
+    public MyHashMap() {
+        table = (Node<K, V>[]) new Node[INITIAL_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        resize();
+        Node<K,V> bucket = table[getBucketIndex(hash(key))];
+        Node<K,V> newNode = new Node<>(key, value, null);
+        if (bucket != null) {
+            while (bucket != null) {
+                if (Objects.equals(bucket.key, key)) {
+                    bucket.value = value;
+                    return;
+                }
+                bucket = bucket.next;
+            }
+            bucket = table[Math.abs(hash(key)) % table.length];
+            while (bucket.next != null) {
+                bucket = bucket.next;
+            }
+            bucket.next = newNode;
+            size++;
+            return;
+        }
+        table[Math.abs(hash(key)) % table.length] = newNode;
+        size++;
```

Comment:
Why do you reset the node here? it's double work, it should already be iterated up to the null node in the previous loop
```suggestion
        node = newNode
        size++;
```